scripts/upgrade

```diff
@@ -34,35 +32,57 @@ upgrade_type=$(ynh_check_app_version_changed)
 #=================================================
 ynh_script_progression --message="Ensuring downward compatibility..."
 
-# Fix is_public as a boolean value
-if [ "$is_public" = "Yes" ]; then
-	ynh_app_setting_set --app=$app --key=is_public --value=1
-	is_public=1
-elif [ "$is_public" = "No" ]; then
-	ynh_app_setting_set --app=$app --key=is_public --value=0
-	is_public=0
-fi
-
 # If final_path doesn't exist, create it
 if [ -z "$final_path" ]; then
 	final_path=/var/www/$app
 	ynh_app_setting_set --app=$app --key=final_path --value=$final_path
 fi
 
-# TODO Not sure if still needed ??
-# admin default value, if not set
-if [ -z "$admin" ]; then
-	admin=$(yunohost user list | grep 'username' -m1 | awk '{print $2}')
-	ynh_app_setting_set --app=$app --key=is_public --value=$is_public
-fi
-
 # language default value, if not set
 if [ -z "$language" ]; then
 	language='en'
 	ynh_app_setting_set --app=$app --key=language --value=$language
 fi
 
-# YunoHost specific configuration, if it isn't exist already
+
+# Cleaning legacy permissions
+admin_user=$(ynh_app_setting_get --app=$app --key=admin)
+
+if [ -n "$admin_user" ]; then
+	# Removing skipped/unprotected_uris under certain conditions, remove the visitors group added during the migration process of 3.7
+	# Remove skipped_uris. If the app was public, add visitors again to the main permission
+	if ynh_permission_has_user --permission=admin --user="$admin_user"
+	then
+		echo "permission admin already exist. Nothing to do"
+	else
+		ynh_permission_create --permission "admin" --allowed "$admin_user"
+	fi
+	# Remove legacy admin setting
+	ynh_app_setting_delete --app=$app --key=admin
+fi
+
+is_public=$(ynh_app_setting_get --app=$app --key=is_public)
+if [ -n "$is_public" ]; then
+	# Remove unprotected_uris
+	ynh_app_setting_delete --app=$app --key=unprotected_uris
+	# Remove protected_uris
+	ynh_app_setting_delete --app=$app --key=protected_uris
+
+	# Removing skipped/unprotected_uris under certain conditions, remove the visitors group added during the migration process of 3.7
+	# Remove skipped_uris. If the app was public, add visitors again to the main permission
+	if ynh_permission_has_user --permission=main --user=visitors
+	then
+		ynh_app_setting_delete --app=$app --key=skipped_uris
+		ynh_permission_update --permission "main" --add "visitors"
+	else
+		ynh_app_setting_delete --app=$app --key=skipped_uris
+	fi
+	ynh_app_setting_delete --app=$app --key=is_public
+fi
```

Comment:
c.f. recent update in example_ynh, this can probably be simplified to : 

```suggestion
# Cleaning legacy permissions
if ynh_legacy_permissions_exists; then
	ynh_legacy_permissions_delete_all

	ynh_app_setting_delete --app=$app --key=is_public
fi
```